core/src/main/java/edu/wpi/grip/core/sources/FileSource.java

```diff
@@ -0,0 +1,87 @@
+package edu.wpi.grip.core.sources;
+
+import edu.wpi.grip.core.Source;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.core.sockets.SocketHint;
+import edu.wpi.grip.core.sockets.SocketHints;
+import edu.wpi.grip.core.util.ExceptionWitness;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.eventbus.EventBus;
+import com.google.inject.assistedinject.Assisted;
+import com.google.inject.assistedinject.AssistedInject;
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * A source for a single file's path. This will only supply the <i>path</i> to the file; it is up
+ * to the operations to resolve the file associated with the path.
+ */
+@XStreamAlias("grip:File")
+public class FileSource extends Source {
```

Comment:
I would disagree with this pattern. I would make this a classifiers source.
Just like we don't expect steps to resolve paths for images that are loaded from the filesystem this too should behave the same.